plugins/transport-nio/src/main/java/org/elasticsearch/http/nio/HttpReadWriteHandler.java

```diff
@@ -77,10 +87,19 @@ public HttpReadWriteHandler(NioHttpChannel nioHttpChannel, NioHttpServerTranspor
     }
 
     @Override
-    public int consumeReads(InboundChannelBuffer channelBuffer) throws IOException {
+    public void channelRegistered() {
```

Comment:
can we add some asserts that this method was called before we call stuff like consumeReads etc?